zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/dynamic/ScaleUpPartitionsTest.java

```diff
@@ -215,4 +251,27 @@ private PlannedOperationsResponse scaleToPartitions(final int desiredPartitionCo
         false,
         false);
   }
+
+  private void awaitScaleUpCompletion(final int desiredPartitionCount) {
+    Awaitility.await("until scaling is done")
+        .atMost(Duration.ofMinutes(2))
+        .catchUncaughtExceptions()
+        .logging()
+        .pollInterval(Duration.ofMillis(500))
+        .untilAsserted(
+            () -> {
+              try {
+                final var topology = clusterActuator.getTopology();
+                assertThat(topology.getRouting()).isNotNull();
+                final var requestHandling = topology.getRouting().getRequestHandling();
+                assertThat(requestHandling).isInstanceOf(RequestHandlingAllPartitions.class);
+                final var allPartitions = (RequestHandlingAllPartitions) requestHandling;
+                assertThat(allPartitions.getPartitionCount()).isEqualTo(desiredPartitionCount);
+              } catch (final Exception e) {
+                System.err.println("Got exception in assertion");
+                e.printStackTrace();
+                throw new RuntimeException(e);
```

Comment:
:x: Why do you have to catch exception here? If you do need it, please use logger to log error instead of System.err.